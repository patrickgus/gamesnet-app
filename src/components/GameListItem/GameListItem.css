.GameListItem {
  display: flex;
  flex-direction: column;
  background-color: rgba(240, 240, 240, 0.6);
  border-radius: 10px;
  margin: 15px auto;
  width: 340px;
  padding: 20px;
}

.GameListItem__header {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.GameListItem__header a {
  color: rgb(36, 36, 36);
}

.GameListItem__link h2 {
  margin: 0 0 20px;
}

.GameListItem__header h3 {
  margin: 15px 0;
}

.GameListItem__link {
  text-align: center;
}

.GameListItem__link img {
  width: 250px;
}

.GameListItem__link:hover {
  color: rgb(71, 71, 71);
  opacity: 0.75;
}

.GameListItem__link h2:hover {
  color: rgb(71, 71, 71);
}

.GameListItem__info {
  display: flex;
  flex-direction: column;
}

.GameListItem__info h4 {
  margin: 5px 0;
  text-align: center;
}

.GameListItem__info p {
  margin: 8px 0;
}

@media screen and (min-width: 680px) {
  .GameListItem {
    width: 480px;
    margin: 15px auto;
  }
}
