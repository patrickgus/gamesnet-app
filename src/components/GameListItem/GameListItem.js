import React, { Component } from "react";
import { Link } from "react-router-dom";
import { Section } from "../../components/Utils/Utils";
import "./GameListItem.css";

export default class GameListItem extends Component {
  render() {
    const { game } = this.props;
    console.log(game);
    return (
      <Section className="GameListItem">
        <header className="GameListItem__header">
          <Link to={`/review/${game.id}`} className="GameListItem__link">
            <h2 className="GameListItem__heading">{game.title}</h2>
          </Link>
          <img src={game.cover} alt={`Game cover art for ${game.title}`} />
          <h4>Avg Rating: {game.avg_rating}</h4>
        </header>
        <div className="gameInfo">
          <h4>Description</h4>
          <p>{game.description}</p>
          <p>Rated: {game.rated}</p>
          <p>Platforms: {game.platforms}</p>
        </div>
      </Section>
    );
  }
}
