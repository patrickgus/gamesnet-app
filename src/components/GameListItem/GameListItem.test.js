import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import GameListItem from "./GameListItem";
import game from "../../stores/game-store";

describe("GameListItem component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <GameListItem game={game} />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the item as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <GameListItem game={game} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
