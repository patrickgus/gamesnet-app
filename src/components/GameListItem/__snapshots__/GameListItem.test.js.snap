// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameListItem component renders the item as expected 1`] = `
<section
  className="Section GameListItem"
>
  <header
    className="GameListItem__header"
  >
    <a
      className="GameListItem__link"
      href="/review/undefined"
      onClick={[Function]}
    >
      <h2
        className="GameListItem__heading"
      />
      <img
        alt="Game cover art for undefined"
        className="GameListItem__cover"
      />
    </a>
    <h3>
      Average Rating: 
    </h3>
  </header>
  <div
    className="GameListItem__info"
  >
    <h4>
      Description
    </h4>
    <p />
    <p>
      <strong>
        Rated:
      </strong>
       
    </p>
    <p>
      <strong>
        Platforms:
      </strong>
       
    </p>
  </div>
</section>
`;
